test/integration/test_api_external_test_environment.py

```diff
@@ -0,0 +1,96 @@
+    generate_root_task,
+    run_task,
+)
+from exasol_integration_test_docker_environment.lib.docker import ContextDockerClient
+from exasol_integration_test_docker_environment.lib.docker.container.utils import (
+    remove_docker_container,
+)
+from exasol_integration_test_docker_environment.lib.models.data.environment_info import (
+    EnvironmentInfo,
+)
+from exasol_integration_test_docker_environment.lib.models.data.environment_type import (
+    EnvironmentType,
+)
+from exasol_integration_test_docker_environment.lib.test_environment.spawn_test_environment import (
+    SpawnTestEnvironment,
+)
+from exasol_integration_test_docker_environment.test.get_test_container_content import (
+    get_test_container_content,
+)
+
+
+@pytest.fixture(scope="module")
+def spawn_test_environment(api_database_module, api_isolation_module):
+    with api_database_module() as db:
+        ext_environment_name = "APISpawnExternalTestExternalEnvironmentTest"
+        task_creator = lambda: generate_root_task(
+            task_class=SpawnTestEnvironment,
+            environment_type=EnvironmentType.external_db,
+            environment_name=ext_environment_name,
+            external_exasol_db_host=db.database_host,
+            external_exasol_db_port=db.ports.database,
+            external_exasol_bucketfs_port=db.ports.bucketfs,
+            external_exasol_ssh_port=db.ports.ssh,
+            external_exasol_db_user=db.db_username,
+            external_exasol_db_password=db.db_password,
+            external_exasol_bucketfs_write_password=db.bucketfs_password,
+            external_exasol_xmlrpc_host=None,
+            external_exasol_xmlrpc_port=443,
+            external_exasol_xmlrpc_user="admin",
+            external_exasol_xmlrpc_password=None,
+            external_exasol_xmlrpc_cluster_name="cluster1",
+            no_test_container_cleanup_after_success=True,
+            no_test_container_cleanup_after_failure=False,
+            reuse_test_container=True,
+            test_container_content=get_test_container_content(),
+            additional_db_parameter=tuple(),
+            docker_environment_variables=tuple(),
+        )
+        ext_environment_info: EnvironmentInfo = run_task(task_creator, 1, None)
+        yield ext_environment_name, ext_environment_info, db.name
+
+        with ContextDockerClient() as docker_client:
+            containers = [
+                c.name
+                for c in docker_client.containers.list()
+                if ext_environment_name in c.name
+            ]
+            remove_docker_container(containers)
+
+
+def test_external_db(spawn_test_environment):
+    ext_environment_name, _, docker_environment_name = spawn_test_environment
+    with ContextDockerClient() as docker_client:
+        containers = [
+            c.name
+            for c in docker_client.containers.list()
+            if docker_environment_name in c.name
+        ]
```

Comment:
maybe  extract that into a function, because you use it a few lines later with simply a different name

Reply:
Done